### PR TITLE
[WIP] Estimator data

### PR DESCRIPTION
**Pull request by zhreshold**
This PR brings multiple improvements to the estimators
- Added Image Classification and Object Detection dataset that's compatible with pandas DataFrame
- Allow CenterNetEstimator to take in train/val data in `fit`
- Default CenterNet construction to one of the COCO pre-trained models, it can significantly improve the performance when training dataset is very small(<500) images
- An example shows how to load various datasets into this particular format: https://colab.research.google.com/drive/1pOnbghr-wetWoMpTEWxW3he8RxQDx_N1#scrollTo=BqvGm0k_oA_I

TODO:
- Remove `dataset` as hyper-parameter
- Allow SSD/Faster/Mask/YOLO to consume train_data/val_data as well
- Allow transfer learning to accelerate the `fit` process

### CODE REVIEW

**Comment by zhreshold** [comment]
@tmwangcas @chongruo 

I'm going to merge this with outstanding issues:

```
************* Module gluoncv.auto.estimators.classification

gluoncv/auto/estimators/classification.py:94:0: W0223: Method '_init_network' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/classification.py:94:0: W0223: Method '_init_trainer' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/classification.py:94:0: W0223: Method '_predict' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/classification.py:94:0: W0223: Method '_resume_fit' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/classification.py:330:4: W0221: Parameters differ from overridden '_fit' method (arguments-differ)

gluoncv/auto/estimators/classification.py:465:4: W0221: Parameters differ from overridden '_evaluate' method (arguments-differ)

************* Module gluoncv.auto.estimators.yolo.yolo

gluoncv/auto/estimators/yolo/yolo.py:32:0: W0223: Method '_init_network' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/yolo/yolo.py:32:0: W0223: Method '_init_trainer' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/yolo/yolo.py:32:0: W0223: Method '_predict' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/yolo/yolo.py:32:0: W0223: Method '_resume_fit' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

************* Module gluoncv.auto.estimators.ssd.ssd

gluoncv/auto/estimators/ssd/ssd.py:34:0: W0223: Method '_init_network' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/ssd/ssd.py:34:0: W0223: Method '_init_trainer' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/ssd/ssd.py:34:0: W0223: Method '_predict' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/ssd/ssd.py:34:0: W0223: Method '_resume_fit' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

************* Module gluoncv.auto.estimators.mask_rcnn.mask_rcnn

gluoncv/auto/estimators/mask_rcnn/mask_rcnn.py:43:0: W0223: Method '_init_network' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/mask_rcnn/mask_rcnn.py:43:0: W0223: Method '_init_trainer' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/mask_rcnn/mask_rcnn.py:43:0: W0223: Method '_predict' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/mask_rcnn/mask_rcnn.py:43:0: W0223: Method '_resume_fit' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

************* Module gluoncv.auto.estimators.faster_rcnn.faster_rcnn

gluoncv/auto/estimators/faster_rcnn/faster_rcnn.py:30:0: W0223: Method '_init_network' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/faster_rcnn/faster_rcnn.py:30:0: W0223: Method '_init_trainer' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/faster_rcnn/faster_rcnn.py:30:0: W0223: Method '_predict' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)

gluoncv/auto/estimators/faster_rcnn/faster_rcnn.py:30:0: W0223: Method '_resume_fit' is abstract in class 'BaseEstimator' but is not overridden (abstract-method)
```

These functions should be implemented accordingly, where for mask rcnn we might need to remove it from the code base and add them later.
